compiler/rustc_resolve/src/late.rs

```diff
@@ -4653,20 +4661,15 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
             let ns = if i + 1 == path.len() { ns } else { TypeNS };
             let res = self.r.partial_res_map.get(&seg.id?)?.full_res()?;
             let binding = self.resolve_ident_in_lexical_scope(seg.ident, ns, None, None)?;
-
-            (res == binding.res()).then_some(seg)
+            (res == binding.clone().res()).then_some((seg, binding))
```

Comment:
```suggestion
            (res == binding.res()).then_some((seg, binding))
```
`LexicalScopeBinding` can be easily made `Copy`, not just `Clone`.